src/components/TeacherViews/Onboarding/Onboarding.test.tsx

```diff
@@ -40,6 +41,28 @@ describe("Onboarding view", () => {
     expect(contactUs).toBeInTheDocument();
     expect(contactUs).toHaveAttribute("href", "/contact-us");
   });
+  it("renders newsletter signup checkbox", () => {
+    renderWithProviders()(<OnboardingView />);
+    expect(
+      screen.getByLabelText(
+        "Sign up to receive helpful content via email. Unsubscribe at any time.",
+      ),
+    ).toBeInTheDocument();
+  });
+  it("should render the Controller component and handle checkbox change", async () => {
+    const { getByRole } = renderWithProviders()(<OnboardingView />);
+
+    const checkbox = getByRole("checkbox", {
+      name: /Sign up to receive helpful content via email. Unsubscribe at any time./i,
+    });
+    expect(checkbox).toBeChecked();
+
+    await act(async () => {
+      fireEvent.click(checkbox);
```

Comment:
pretty sure fireEvent wraps in `act` internally - were you seeing warnings without it?

Reply:
Yeah was getting loads of console warnings so chucked this around it, happy to remove / ignore them?

Reply:
nah definitely don't ignore them, I'd maybe try quickly refactoring to the `await`/`findBy` style of the other tests in here just for clarity, but if it is still having issues this is fine

Reply:
Refactored ✅